### PR TITLE
Flag for stats

### PR DESCRIPTION
**Pull request by Ritvik1sharma**

### CODE REVIEW

**Comment by weiya711** [comment]
It seems that your statistics flag isn't being used anywhere and doesn't actually disable the computation to get the statistics anywhere in `sam/sim/src` @Ritvik1sharma

**Review by weiya711 — CHANGES_REQUESTED** [review]
As said earlier, it seems the statistics flag for each class object is passed in but not being used anywhere in `sam/sim/src`. How is it actually being used to disable or enable anything? Am I missing something? 

The statistics flag is also only added to the final-apps directory and not all of the other apps directories. Also is statistics by default false or true?

**Comment by Ritvik1sharma** [comment]
Stats sjhould be there now. By default no stats are reported.

File: sam/sim/src/rd_scanner.py [hunk 1]
```diff
@@ -169,23 +166,24 @@ def __init__(self, crd_arr=[], seg_arr=[], skip=True, **kwargs):
...
-        self.fiber_behind_cnt = 0
+        if self.get_stats:
+            self.unique_refs = []
+            self.unique_crds = []
+            self.total_outputs = 0
+            self.elements_skipped = 0
+            self.skip_cnt = 0
+            self.intersection_behind_cnt = 0
+            self.fiber_behind_cnt = 0
+        self.stop_count = 0
```
Comment (weiya711) [inline sam/sim/src/rd_scanner.py]: I think this should be under self.get_stats?

Comment (Ritvik1sharma) [reply sam/sim/src/rd_scanner.py]: You mean stop_count = 0 line? 
It is not supposed to be, it is used to set the ;level of the stop counter generated by the read scanner. Hence is needed outside.

Comment (weiya711) [reply sam/sim/src/rd_scanner.py]: @Ritvik1sharma the stop_count from what i've seen is only used in the dictionary

Comment (Ritvik1sharma) [reply sam/sim/src/rd_scanner.py]: Yes I got that. I had corrected this in the pull request.

**Review by weiya711 — APPROVED** [review]
Everything has been resolved

**Pull request merged**